src/integritysetup/integritysetup-generator.c

```diff
@@ -0,0 +1,226 @@
+
+        char *save_ptr = NULL, *match = NULL;
+
+        assert(o);
+        assert(token);
+
+        match = strtok_r(token, "=", &save_ptr);
+        if (match) {
+                if (strlen(match) > 1)
+                        fprintf(o, "-");
+
+                fprintf(o, "-%s ", match);
+                match = strtok_r(NULL, "=", &save_ptr);
+                if (match)
+                        fprintf(o, "%s ", match);
+        }
+}
+
+static int options_string(
+                char *options,
+                char **options_formatted) {
+
+        char *buff = NULL, *save_ptr = NULL, *token = NULL;
+        size_t len = 0;
+        int r;
+        FILE *out = NULL;
+
+        if (!(options && options_formatted  && *options_formatted == NULL && strlen(options) > 0)) {
+               return -EINVAL;
+        }
+
+        out = open_memstream(&buff, &len);
+        if (!out)
+                return log_oom();
+
+        token = strtok_r(options, ",", &save_ptr);
+        while (token != NULL) {
+                option(out, token);
+                token = strtok_r(NULL, ",", &save_ptr);
+        }
+
+        r = fclose(out);
+        if (r != 0)
+                return log_error_errno(r, "fclose failure on opened memstream: %m");
+
+        if (len > 0) {
+                *options_formatted = buff;
+        }
+        return 0;
+}
+
+static int create_disk(
+                const char *name,
+                const char *device,
+                char *options) {
+
+        _cleanup_free_ char *n = NULL, *dd = NULL, *e = NULL, *name_escaped = NULL, *options_convert = NULL;
+        _cleanup_fclose_ FILE *f = NULL;
+        int r;
+        char *dmname = NULL;
+
+        assert(name);
+        assert(device);
+
+        name_escaped = specifier_escape(name);
+        if (!name_escaped)
+                return log_oom();
+
+        e = unit_name_escape(name);
+        if (!e)
+                return log_oom();
+
+        r = unit_name_build("systemd-integritysetup", e, ".service", &n);
+        if (r < 0)
+                return log_error_errno(r, "Failed to generate unit name: %m");
+
+        r = unit_name_from_path(device, ".device", &dd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to generate unit name: %m");
+
+        r = generator_open_unit_file(arg_dest, NULL, n, &f);
+        if (r < 0)
+                return r;
+
+        if (options) {
+                r = options_string(options, &options_convert);
+                if (r < 0)
+                        return r;
+        }
+
+        fprintf(f,
+                "[Unit]\n"
+                "Description=Integrity Setup for %%I\n"
+                "Documentation=man:integritytab(5) man:systemd-integritysetup-generator(8) man:systemd-integritysetup@.service(8)\n"
+                "SourcePath=%s\n"
+                "DefaultDependencies=no\n"
+                "IgnoreOnIsolate=true\n"
+                "After=systemd-udevd-kernel.socket\n"
```

Comment:
I have the suspicion this should also include `After=integritysetup-pre.target`. We have the same for cryptseutp/veritysetup